ui/src/tool-panel/tool-panel.tsx

```diff
@@ -185,15 +198,15 @@ const ToolPanel: React.FC<ToolPanelProps> = ({
                         }
                     />
                 ),
-            };
+            } as ToolPanelTab;
```

Comment:
En keksinyt miksi, mutta TypeScript pakottaa nämä castit. Ilman niitä tyypit eivät osu yhteen, vaikka kyllä ne oikeasti osuvatkin

Reply:
toimisko, jos laittais planTabsin tyypiksi ToolPanelTab[]? Vähän sama asia kylläkin

Reply:
Kokeilin tuota niin kääntäjä valitti että "Type 'string' is not assignable to type '"LOCATION_TRACK" | "SWITCH" | "KM_POST" | "REFERENCE_LINE" | "TRACK_NUMBER" | "GEOMETRY_ALIGNMENT" | "GEOMETRY_PLAN" | "GEOMETRY_KM_POST" | "GEOMETRY_SWITCH"'"

Tuon perusteella se ei tajua että nuo välimuuttujan tyypit on noita ToolPanelTab ja siksi luulee että tuo kenttä olisi vain string. Castin sijaan tosiaan tuo tiandun ehdottama olisi nätimpi jos toimii.